lib/rules/selector-max-compound-selectors/__tests__/index.mjs

```diff
@@ -384,3 +384,82 @@ testRule({
 		},
 	],
 });
+
+// Testing ignoreSelectors
+testRule({
+	ruleName,
+	config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/] }],
+
+	accept: [
```

Comment:
_if the status is changed to ready to implement_

add a nesting example
i.e. would `:global(.foo)` count as 1 or 0?
If it's 1, then similar selectors like `:not` will require _unpacking_ if ignored

Reply:
Agree. We want test cases using `:not()` like this:

https://github.com/stylelint/stylelint/blob/da72434387fde008df95a13eb4f8ecbb22b8c68c/lib/rules/selector-max-compound-selectors/__tests__/index.mjs#L123

Reply:
I'm not sure which tests I should add. Could you tell me (high-level) the code you would like to be tested as accepted and not accepted?

Reply:
It depends on my question :

> `:global(.foo)` count as 1 or 0?

A If it's currently 0 then we don't need a new test case.
B If it's not then you gotta at least check that the behaviour is coherent when unpacking the _complex real selector list_ of the ignored selector.
e.g. (if B) `:not(.a, div)` should be 2 based on

https://github.com/stylelint/stylelint/blob/b6446093297c9a6a147c3c967067f06c12d76ab1/lib/rules/selector-max-compound-selectors/README.md?plain=1#L60

Reply:
`:global()` is non-standard, so let's ignore it here.

I think it's enough to add cases for a functional pseudo-class e.g. `:not()`:

Accept:

```css
:not(.foo ::v-deep .bar) {}
```

Reject:

```css
:not(.foo ::v-deep .bar .baz) {}
```

Reply:
> Reject:
> 
> ```css
> :not(.foo ::v-deep .bar .baz) {}
> ```

If it's for `2` with an ignore of `::v-deep`, it's an incorrect test.
If it's for `2` with an ignore of `:not`, it's a correct test.

Reply:
I meant using `config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/] }]` at L31. So, I need clarification on why the following is incorrect.

> If it's for `2` with an ignore of `::v-deep`, it's an incorrect test.

Reply:
> I meant using `config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/] }]` at L31. So, I need clarification on why the following is incorrect.
> 
> > If it's for `2` with an ignore of `::v-deep`, it's an incorrect test.

It was not immediately clear to me until I read :

> `:not()` is considered one compound selector irrespective to the complexity of the selector inside it. The rule does process that inner selector, but does so separately, independent of the main selector.

so

> Reject:
> 
> ```css
> :not(.foo ::v-deep .bar .baz) {}
> ```

wasn't 4 but 1 *and* 3.
i.e. two separate evaluations

My bad, I missed that warning in the documentation. The test is correct.
Could we add a `> [!WARNING]` to

https://github.com/stylelint/stylelint/blob/b6446093297c9a6a147c3c967067f06c12d76ab1/lib/rules/selector-max-compound-selectors/README.md?plain=1#L17

?

Reply:
> Could you tell me (high-level) the code you would like to be tested as accepted and not accepted?

@FloEdelmann

`config: [2, { ignoreSelectors: [':not'] }]`

Accept:

```css
p a :not(.foo .bar) {}
```

Reject:

```css
p a :not(.foo .bar .baz) {}
```

i.e. if `not` is ignored, does the inner selector gets evaluated so that it can be rejected or should you bail and get rid of it completely and hence

```css
p a :not(.foo .bar .baz) {}
```

should pass too?

Reply:
> Could we add a `> [!WARNING]` to […]?

I added it in [`0cba157`](https://github.com/stylelint/stylelint/pull/7544/commits/0cba1578168089aee3d931e47a4fad64cb066ffa). I needed to add a `<!-- prettier-ignore -->` comment above it, otherwise Prettier would merge the two lines together into one, apparently because the second line starts with inline code (`` ` ``).

> […] hence `p a :not(.foo .bar .baz) {}` should pass too?

I don't think ignoring `:not` should also ignore contents of it.

---

I changed the implementation to support ignoring `:not` and updated the tests and docs accordingly in [`e174fc5`](https://github.com/stylelint/stylelint/pull/7544/commits/e174fc57a6949389cec58773879cef5e146a7ff2).

Reply:
> I don't think ignoring :not should also ignore contents of it.

Ill add that to OP.